uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,453 @@ define([
+          vm.departments = departments;
+        });
+    }
+
+    /**
+     * Loads all leave requests
+     *
+     * @param {string} type - load leave requests for the either the filter or the table
+     * @return {Promise}
+     */
+    function loadLeaveRequest(type) {
+      var filterByStatus = type !== 'filter',
+        pagination = type === 'filter' ? {} : vm.pagination,
+        returnFields = type === 'filter' ? {
+            return: ['status_id']
+          } : {};
+
+      return LeaveRequest.all(leaveRequestFilters(filterByStatus), pagination, null, returnFields)
+        .then(function (leaveRequests) {
+          vm.leaveRequests[type] = leaveRequests;
+        });
+    }
+
+    /**
+     * Loads the level types option values
+     *
+     * @return {Promise}
+     */
+    function loadLevelTypes() {
+      return OptionGroup.valuesOf('hrjc_level_type')
+        .then(function (levels) {
+          vm.levelTypes = levels;
+        });
+    }
+
+    /**
+     * Loads the locations option values
+     *
+     * @return {Promise}
+     */
+    function loadLocations() {
+      return OptionGroup.valuesOf('hrjc_location')
+        .then(function (locations) {
+          vm.locations = locations;
+        });
+    }
+
+    /**
+     * Returns the filter object for leave request api
+     *
+     * @param {boolean} filterByStatus - if true then leave request api will be filtered using
+     * selected leave request status in the left navigation bar, which would be used to show the
+     * numbers of different statuses
+     * @return {Object}
+     */
+    function leaveRequestFilters(filterByStatus) {
+      var filters = vm.filters.leaveRequest;
+
+      return {
+        managed_by: vm.contactId,
+        type_id: filters.selectedAbsenceTypes ? filters.selectedAbsenceTypes.id : null,
+        status_id: prepareStatusFilter(filterByStatus),
+        from_date: {
+          from: filters.selectedPeriod.start_date
+        },
+        to_date: {
+          to: filters.selectedPeriod.end_date
+        },
+        contact_id: prepareContactID()
+      };
+    }
+
+    /**
+     * Loads the regions option values
+     *
+     * @return {Promise}
+     */
+    function loadRegions() {
+      return OptionGroup.valuesOf('hrjc_region')
+        .then(function (regions) {
+          vm.regions = regions;
+        });
+    }
+
+    /**
+     * Loads the status option values
+     *
+     * @return {Promise}
+     */
+    function loadStatuses() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
+        .then(function (statuses) {
+          vm.leaveRequestStatuses = statuses.concat(vm.leaveRequestStatuses);
+        });
+    }
+
+    /**
+     * Returns the contact ID to be used for leave request api
+     *
+     * @return {Object}
+     */
+    function prepareContactID() {
+      if (vm.filters.leaveRequest.selectedUsers) {
+        return vm.filters.leaveRequest.selectedUsers.contact_id;
```

Comment:
Ok I think the name here is misleading, as `selectedUsers` gives the idea that it is an array of multiple users, while in fact it's a single object (one user). 

Since we're on the subject of naming, wouldn't this make much more sense to have it on the `vm.filters.contact` group? After all it is grouped with the rest of the contact filters

Something simple like `vm.filters.contact.id` would be immediately understandable